internal/sync/lifecycle_test.go

```diff
@@ -146,15 +147,39 @@ func TestLifecycleOnce(t *testing.T) {
 				StartAction{ExpectedState: Running},
 				ConcurrentAction{
 					Actions: []LifecycleAction{
-						StopAction{ExpectedState: Stopped, Wait: 20 * time.Millisecond},
+						StopAction{ExpectedState: Stopped, Wait: 15 * time.Millisecond},
 						GetStateAction{ExpectedState: Stopping},
+						GetStateAction{ExpectedState: Stopped},
 					},
 					Wait: 10 * time.Millisecond,
 				},
 			},
 			expectedFinalState: Stopped,
 		},
 		{
+			msg: "Delayed stop, wait for stopping",
+			actions: []LifecycleAction{
+				StartAction{ExpectedState: Running},
+				ConcurrentAction{
+					Actions: []LifecycleAction{
+						Actions{
+							StartAction{ExpectedState: Running},
+							WaitAction(20 * time.Millisecond),
+							StopAction{ExpectedState: Stopped, Wait: 20 * time.Millisecond},
+						},
+						Actions{
+							WaitForStoppingAction,
+							WaitAction(10 * time.Millisecond),
+							ExactStateAction{ExpectedState: Stopping},
+							WaitAction(20 * time.Millisecond),
```

Comment:
I could use a quick explanation here - does this mean that between 10 and 20 ms, the action is expected to be Stopping, and then >=20 ms, the action is expected to be Stopped? This seems like it could be flaky.